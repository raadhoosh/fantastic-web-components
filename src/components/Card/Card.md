
Card example:

```js
<Card>
    <CardHeader >
        Pizza
    </CardHeader>
    <CardBody>
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
    </CardBody>
    <CardFooter>
        <Button>
            Accept
        </Button> 
    </CardFooter>
</Card>
```

Card example Background Color:

```js
<Card>
    <CardHeader primary>
        Pizza
    </CardHeader>
    <CardBody>
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
    </CardBody>
    <CardFooter primary>
        <Button secondary>
            Accept
        </Button> 
    </CardFooter>
</Card>

<br/>
<Card>   
    <CardBodyImg/>
    <CardFooter>
       John
       <p>Architect &amp; Engineer</p>
    </CardFooter>
</Card>
<br/>
<Card>
    <CardHeader success>
       Header
    </CardHeader>
    <CardBody>
      Some text.. Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
    </CardBody>
    <CardFooter success>
       Footer
    </CardFooter>
</Card>


```