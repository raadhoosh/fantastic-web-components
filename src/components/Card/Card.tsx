import styled from '../../common/styled';
import * as React from 'react';
export interface IProps {
    /** This is a primary button it out!. */
    primary?: boolean;
    /** This is a secondary button it out!. */
    secondary?: boolean;
    /** This is a success button it out!. */
    success?: boolean;
    /** This is a info button it out!. */
    info?: boolean;
    /** This is a warning button it out!. */
    warning?: boolean;
    /** This is a danger button it out!. */
    danger?: boolean;
    /** This is a light button it out!. */
    light?: boolean;
    /** This is a dark button it out!. */
    dark?: boolean;
    /** This is a text color button it out! */
    children?: any;
    /** This is a Text Color button it out!. */
    color?: string;
    /** This is a margin button it out!. */
    margin?: string;
    radius?: string;
    bgColor?: string;
    borderColor?: string;
    /** This is a noShadow button it out!. */
    noShadow?: boolean;
    theme?: any;
    size?: 'sm' | 'lg' | 'md';
}
const EmCard = styled('div')(
    (props: (IProps)) => {
        const base = {
            width: '499px',
            border: '1px solid',
            borderRadius: '.25rem',
            borderColor: 'rgba(0,0,0,.125)',
            backgroundColor: 'transparent',
            boxShadow: '0 2px 5px 0 rgba(0,0,0,0.01), 0 2px 10px 0 rgba(0,0,0,0.15)',
            color: '#000',
            '*, ::after, ::before': {
                'box-sizing': 'border-box',
            }
        };
        if (props.radius) {
            base.borderRadius = props.radius;
        }
        if (props.bgColor) {
            if (props.theme.color[props.bgColor]) {
                base.backgroundColor = props.theme.color[props.bgColor];
            } else {
                base.backgroundColor = props.bgColor;
            }
        }
        const color = (props.primary && 'primary') ||
            (props.secondary && 'secondary') ||
            (props.success && 'success') ||
            (props.info && 'info') ||
            (props.danger && 'danger') ||
            (props.warning && 'warning') ||
            (props.light && 'light') ||
            (props.dark && 'dark') ||
            ('');
        if (color !== '') {
            base.backgroundColor = props.theme.color[color];
        }
        if (props.borderColor) {
            if (props.theme.color[props.borderColor]) {
                base.borderColor = props.theme.color[props.borderColor];
            } else {
                base.borderColor = props.borderColor;
            }
        }
        base.color = props.color ? props.color : base.color;
        return base;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const Card = (props: IProps) => {

    return (
        <EmCard {...props}> {props.children} </EmCard>
    );
};

export default Card;