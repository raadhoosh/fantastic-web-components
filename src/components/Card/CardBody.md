CardBody example:

```js
    <CardBody>
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
        🍕🍕🍕🍕🍕🍕🍕🍕🍕
    </CardBody>
```