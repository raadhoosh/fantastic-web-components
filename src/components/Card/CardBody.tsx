import styled from '../../common/styled';
import * as React from 'react';
export interface IProps {
    /** This is a text color button it out! */
    children?: any;
    /** This is a Background Color button it out!. */
    backgroundColor?: string;
    /** This is a Text Color button it out!. */
    color?: string;
    /** This is a margin button it out!. */
    margin?: string;
    radius?: string;
    bgColor?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark',
    borderColor?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark',
    /** This is a noShadow button it out!. */
    noShadow?: boolean;
    theme?: any;
    size?: 'sm' | 'lg' | 'md';
}
const EmCardBody = styled('div')(
    (props: (IProps)) => {
        const base = {
            flex: '1 1 auto',
            padding: '.75rem .75rem',
            '*, ::after, ::before': {
                'box-sizing': 'border-box',
            }
        };
        return base;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const CardBody = (props: IProps) => {

    return (
        <EmCardBody {...props}> {props.children} </EmCardBody>
    );
};

export default CardBody;
