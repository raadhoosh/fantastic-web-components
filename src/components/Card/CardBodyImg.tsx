import styled from '../../common/styled';
import * as React from 'react';
import imgSrc from './img_avatar3.png';

export interface IProps {
    /** This is a text color button it out! */
    children?: any;
    /** This is a Background Color button it out!. */
    backgroundColor?: string;
    /** This is a Text Color button it out!. */
    color?: string;
    /** This is a margin button it out!. */
    margin?: string;
    radius?: string;
    bgColor?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark',
    borderColor?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'danger' | 'light' | 'dark',
    /** This is a noShadow button it out!. */
    noShadow?: boolean;
    theme?: any;
    size?: 'sm' | 'lg' | 'md';
}
const EmCardBodyImg = styled('div')(
    (props: (IProps)) => {
        const base = {
            width: '200px',
            padding: '0',
            '*, ::after, ::before': {
                'box-sizing': 'border-box',
            }
        };
        return base;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const CardBodyImg = (props: IProps) => {

    return (
        <EmCardBodyImg {...props}>
        <img src={imgSrc}/>
        </EmCardBodyImg>
    );
};

export default CardBodyImg;
