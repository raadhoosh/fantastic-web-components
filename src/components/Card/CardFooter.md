CardFooter example:

```js
<CardFooter><Button>Accept</Button> </CardFooter>
```