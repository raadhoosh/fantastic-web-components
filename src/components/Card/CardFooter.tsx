import styled from '../../common/styled';
import * as React from 'react';
export interface IProps {
    /** This is a primary button it out!. */
    primary?: boolean;
    /** This is a secondary button it out!. */
    secondary?: boolean;
    /** This is a success button it out!. */
    success?: boolean;
    /** This is a info button it out!. */
    info?: boolean;
    /** This is a warning button it out!. */
    warning?: boolean;
    /** This is a danger button it out!. */
    danger?: boolean;
    /** This is a light button it out!. */
    light?: boolean;
    /** This is a dark button it out!. */
    dark?: boolean;
    /** This is a text color button it out! */
    children?: any;
    /** This is a Background Color button it out!. */
    backgroundColor?: string;
    /** This is a Text Color button it out!. */
    color?: string;
    /** This is a margin button it out!. */
    margin?: string;
    radius?: string;
    bgColor?: string;
    borderColor?: string;
    /** This is a noShadow button it out!. */
    noShadow?: boolean;
    theme?: any;
    size?: 'sm' | 'lg' | 'md';
}
const EmCardFooter = styled('div')(
    (props: (IProps)) => {
        const base = {
            borderTop: '1px solid',
            padding: '.75rem .75rem',
            marginTop: 0,
            backgroundColor: 'transparent',
            borderRadius: 'none',
            borderTopColor: 'transparent',
            ':first-child': {
                borderRadius: 'calc(.25rem - 1px) calc(.25rem - 1px) 0 0'
            }
        };
        if (props.radius) {
            base.borderRadius = props.radius;
        }
        if (props.bgColor) {
            if (props.theme.color[props.bgColor]) {
                base.backgroundColor = props.theme.color[props.bgColor];
            } else {
                base.backgroundColor = props.bgColor;
            }
        }
        const color = (props.primary && 'primary') ||
            (props.secondary && 'secondary') ||
            (props.success && 'success') ||
            (props.info && 'info') ||
            (props.danger && 'danger') ||
            (props.warning && 'warning') ||
            (props.light && 'light') ||
            (props.dark && 'dark') ||
            ('');
        if (color !== '') {
            base.backgroundColor = props.theme.color[color];
        }
        if (props.borderColor) {
            if (props.theme.color[props.borderColor]) {
                base.borderTopColor = props.theme.color[props.borderColor];
            } else {
                base.borderTopColor = props.borderColor;
            }
        }
        return base;
    }
);
/**
 * General component description in JSDoc format. Markdown is *supported*.
 */
const CardFooter = (props: IProps) => {

    return (
        <EmCardFooter {...props}> {props.children} </EmCardFooter>
    );
};
export default CardFooter;
