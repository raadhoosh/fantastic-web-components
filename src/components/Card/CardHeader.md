CardHeader example:

```js
<CardHeader >Title Header</CardHeader>
<CardHeader bgColor="primary">Title Header</CardHeader>
```