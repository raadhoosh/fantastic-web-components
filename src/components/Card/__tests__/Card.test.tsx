import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createMatchers } from 'jest-emotion';
import * as emotion from 'emotion';

// Add the custom matchers provided by 'jest-emotion'
expect.extend(createMatchers(emotion));

import Card from '../Card';
import mainTheme from '../../../common/mainTheme';

test('renders with correct styles', () => {
    const tree = renderer.create(<Card theme={mainTheme} primary>hi test</Card>).toJSON();
    expect(tree).toMatchSnapshot();
});
