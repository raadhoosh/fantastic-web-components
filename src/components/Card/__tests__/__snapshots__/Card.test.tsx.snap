// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with correct styles 1`] = `
<div
  className="css-104y8oi"
>
  hi test
</div>
`;
